src/tokenizer/mod.rs

```diff
@@ -561,6 +562,19 @@ impl<'a> Tokenizer<'a> {
             self.gen_punct(Punct::Caret)
         }
     }
+    fn hash(&mut self) -> Res<RawItem> {
+        // hashbang comment can only appear at the start
+        if self.current_start == 0 && self.stream.next_char() == Some('!') {
```

Comment:
This should probably be updated to 

`self.current_start == 0 && self.look_ahead_byte_matches('!')`